SQA_AT/src/main/java/general/BaseGeneralPage.java

```diff
@@ -1,15 +1,38 @@
 package general;
 
 import driver.DriverRepository;
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebElement;
 import org.openqa.selenium.remote.RemoteWebDriver;
 import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
 
 public class BaseGeneralPage {
 
+    private static final int WAIT_TWO_SEC = 2;
+
     protected RemoteWebDriver driver;
 
+    protected WebDriverWait wait;
+
     public BaseGeneralPage() {
         driver = DriverRepository.DRIVERS.get();
+        wait = new WebDriverWait(driver, Duration.ofSeconds(WAIT_TWO_SEC));
         PageFactory.initElements(driver, this);
```

Comment:
initElements(driver, this); we should do on a page level...
 pls check here more details "Sample Project Structure for Page Factory"
 https://www.browserstack.com/guide/page-object-model-in-selenium